pkg/geo/geomfn/linestring.go

```diff
@@ -252,3 +253,60 @@ func removePoint(lineString *geom.LineString, index int) (*geom.LineString, erro
 
 	return lineString.SetCoords(coords)
 }
+
+// LineSubString returns a LineString being a substring by start and end
+func LineSubString(g geo.Geometry, start, end float64) (geo.Geometry, error) {
+
+	if start < 0 || start > 1 || end < 0 || end > 1 {
+		return g, pgerror.Newf(pgcode.InvalidParameterValue,
+			"start and and must be within 0 and 1")
+	}
+	if start > end {
+		return g, pgerror.Newf(pgcode.Internal,
+			"2nd arg must be smaller then 3rd arg")
+	}
+
+	if start == end {
+		return LineInterpolatePoints(g, start, false)
+	}
+
+	lineT, err := g.AsGeomT()
+	if err != nil {
+		return g, err
+	}
+	lineString, ok := lineT.(*geom.LineString)
+	if !ok {
+		return g, pgerror.Newf(pgcode.InvalidParameterValue,
+			"first parameter has to be of type LineString")
+	}
+
+	var newCoords, tmpCoords []float64
+	var stop bool
+	for i := 0; i < len(lineString.FlatCoords())-1 && !stop; i += 2 {
```

Comment:
this will not futureproof us for 3D/4D coords. can we perhaps use the `.Coords()` function here?